fix: fix sysbox resources

charlielye opened the pull request:
Seems like we regressed ourselves into a state where we couldn't run all the tests on the mainframe without hitting resource limits. e.g. running `yarn test` in yarn-project hits ~22k threads peak. This in itself needs investigating.
But, the container limit is 65k, so what was the issue?
Systemd has it's own limits - by default 15% the total, which is why bumping container pids helped, but not as much as i expected. This now ensures everything is maxed up the sysbox container limit.
As I've rebuilt the amis im also removing the ensure_zig stuff as its in the container now (as well as ldid).

I've had to fix a load of issues introduced by the foundry version bump. I'm unsure why the version was bumped when the existing version seemed to be working...

edit: An investigation into the 22k threads was a combo of the cursed rayon and also newly discovered Tokio thread pool that for some inexplicable reason is in kevs kzg lib. Controlled down with TOKIO_WORKER_THREADS=1. A full parallel run of yarn-project tests now only seems to peak about 1k additional threads.

ludamad commented:
Can we add these tools to our bootstrap tool checker?

ludamad reviewed (approved):
lgtm other than comment about catching these not being installed early

charlielye committed:
fix: fix sysbox resources

Seems like we regressed ourselves into a state where we couldn't run all the tests on the mainframe without hitting resource limits. e.g. running `yarn test` in yarn-project hits ~22k threads peak. This in itself needs investigating.
But, the container limit is 65k, so what was the issue?
Systemd has it's own limits - by default 15% the total, which is why bumping container pids helped, but not as much as i expected. This now ensures everything is maxed up the sysbox container limit.
As I've rebuilt the amis im also removing the ensure_zig stuff as its in the container now (as well as ldid).

I've had to fix a load of issues introduced by the foundry version bump. I'm unsure why the version was bumped when the existing version seemed to be working...
